infra/circleci/orbs.md

```diff
@@ -0,0 +1,209 @@
+```
+
+From then on, every time this CircleCI job runs, a slack notification will appear in the specified channel.
+
+Using an orb's jobs versus its commands is slightly different. An example would be using `circleci/slack`'s `approval-notification` which notifies a channel of a pending approval through the `approval` command.
+
+Instead of adding a step to a job, you add the job to a workflow
+
+```yaml
+workflows:
+  your-workflow:
+    jobs:
+      - slack/approval-notification:
+          message: < message >
+          webhook: < slack verification token >
+```
+
+For concrete examples of the usage of this orb, check out [legendary-waddle-dev](https://github.com/trussworks/legendary-waddle-dev/blob/master/.circleci/config.yml).
+
+## Creating Orbs
+
+Unlike using orbs, the process around creating orbs can be tricky. There's plenty of [documentation](https://circleci.com/docs/2.0/creating-orbs/) to go through but I'll provide you a condensed version with some Truss pizzazz.
+
+### Requirements
+
+- [CircleCI CLI](https://circleci.com/docs/2.0/orb-author-cli/#configure-the-circleci-cli) is installed and set up on your machine
+
+  ```sh
+  brew install circleci
+  ```
+
+- You have the owner role within the Trussworks GitHub org or can snag some time with someone who is an owner
+
+### GitHub
+
+Your orb's source code needs to be housed somewhere. We'll be using GitHub to start as it's where our organization currently holds internal repositories and our CircleCI namespace is linked to GitHub.
+
+A CircleCI orb repository should
+
+- have the name prefixed with "orb-" (ie. `orb-ecr-image-scan-findings`)
+- add the MIT license
+- be set to public
+- include a README describing what the orb does
+- use pre-commit hooks to check the yaml and lint markdown
+- possess CircleCI configuration for validation
+
+### Setting up the orb
+
+Now that we have a GitHub repo and a name for our orb, we can create the orb itself.
+
+In your terminal, let's reserve the name of the orb via
+
+```sh
+circleci orb create trussworks/<name of the orb>
+```
+
+Within our GitHub repo, create a file called `orb.yml` with the following content
+
+```yaml
+version: 2.1
+
+description: >
+ < explanation of what your orb does >
+
+display:
+    home_url: https://truss.works/
+    source_url: <github repo url>
+
+# these are steps you can add to a job
+commands:
+    placeholder-command:
+        description: < explanation of what this specific command does >
+        parameters: < optional if you need to pass arguments in>
+            <command parameter one>:
+                type: < type of argument >
+                description: < explanation of what is this argument>
+        steps:
+            - run:
+                command: < the actual logic behind what your orb does>
+
+# the environment the job will run in
+executors:
+    < placeholder-executor >:
+        # can use a machine or some other environment
+        docker:
+            - image: < placeholder-image >
```

Comment:
This is more of a mechanical question, but when I write a CircleCI config I specify an executor as well -- does the orb run in the executor for the orb config, or the executor for the config I write in my `config.yml`? Does an orb *need* an executor?

Reply:
To your more general comment:
I was personally torn between a generic example and a more specific example. I definitely stand to benefit from a specific example myself but I thought that a compromise of a general example while pointing to the orb we do have published would be sufficient enough.

To the more specific question:
I don't believe an orb always needs an executor. 

If you're using an orb's job, you would be using the executor (the one in the orb config) specified in that job. Like in L159. 

If you're using an orb's command, the orb will use the executor specified in the config it was imported to. In this case, you're `config.yml`.